src/lib/stellar/transactions.js

```diff
@@ -331,3 +332,58 @@ export async function createPathPaymentStrictReceiveTransaction({
         network_passphrase: networkPassphrase,
     }
 }
+
+/**
+ * Constructs and returns a Stellar transaction for transferring assets to a contract or account.
+ * @async
+ * @function createContractTransferTransaction
+ * @param {Object} opts Options object
+ * @param {string} opts.source Public Stellar address to use as the source account of the transaction
+ * @param {string} opts.destination Public Stellar address or contract ID to receive the transfer
+ * @param {string} opts.amount Amount of the asset to transfer
+ * @param {string} opts.asset Asset to be transferred (example: USDC:GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5)
+ * @returns {Promise<TransactionResponse>} Object containing the relevant network passphrase and the built transaction envelope in XDR base64 encoding, ready to be signed and submitted
+ */
+export async function createContractTransferTransaction({ source, destination, amount, asset }) {
+    const server = new Server(horizonUrl);
+    const sourceAccount = await server.loadAccount(source);
+
+    const transaction = new TransactionBuilder(sourceAccount, {
+        networkPassphrase: networkPassphrase,
+        fee: maxFeePerOperation,
+    });
+
+    if (StrKey.isValidContract(destination)) {
+        // Transfer to a contract
+        const [assetCode, assetIssuer] = asset.split(':');
+        const contract = new Contract(destination);
+        const amountBigInt = BigInt(amount);
+        const transferOp = contract.call(
+            "transfer",
+            Address.fromString(source).toScVal(),
+            Address.fromString(destination).toScVal(),
+            xdr.ScVal.scvI128(new xdr.Int128Parts({
+                lo: xdr.Uint64.fromString(amountBigInt.toString(16).padStart(16, '0').slice(-16)),
+                hi: xdr.Int64.fromString(amountBigInt.toString(16).slice(0, -16) || '0')
+            })),
+            xdr.ScVal.scvSymbol(assetCode)
+        );
+        transaction.addOperation(transferOp);
+    } else {
+        // Regular payment to an account
+        let sendAsset = asset === 'native' ? Asset.native() : new Asset(asset.split(':')[0], asset.split(':')[1]);
+        transaction.addOperation(
+            Operation.payment({
+                destination: destination,
+                asset: sendAsset,
+                amount: amount.toString()
+            })
+        );
+    }
+
+    const builtTransaction = transaction.setTimeout(standardTimebounds).build();
```

Comment:
for contract invocations, you'll need to simulate the transaction. The rpc server has [the `prepareTransaction()` function](https://stellar.github.io/js-stellar-sdk/module-rpc.Server.html#prepareTransaction) that will make this process pretty easy on you.

Reply:
Okay sir I’m on it please